### PR TITLE
Major refactoring to support new keystore modes. Some functionality removed (package removal).

### PR DESCRIPTION
**Pull request by atrullmdsol**

### CODE REVIEW

**Comment by atrullmdsol** [comment]
This work really needs nice tests.

File: providers/default.rb [hunk 1]
```diff
@@ -49,99 +49,273 @@ def whyrun_supported?
...
+        puts "Encrypting device item for #{name}"
+        encrypted_new_deviceitem = Chef::EncryptedDataBagItem.encrypt_data_bag_item(new_deviceitem, encrypted_data_bag_secret)
+        deviceitem = Chef::DataBagItem.from_hash(encrypted_new_deviceitem)
+      else
+        # Unencrypted databag item.
+        deviceitem = Chef::DataBagItem.new
+        deviceitem.raw_data = new_deviceitem
+      end 
+      deviceitem.data_bag(keystore_databag_name)
+      deviceitem.save
```
Comment (atrull) [inline providers/default.rb]: This bit is 403ing when the databag exists and the clients have update perms on that databag.

File: providers/default.rb [hunk 2]
```diff
@@ -49,195 +49,334 @@ def whyrun_supported?
...
+        puts "Encrypting device item for #{name}"
+        encrypted_new_deviceitem = Chef::EncryptedDataBagItem.encrypt_data_bag_item(new_deviceitem, encrypted_data_bag_secret)
+        deviceitem = Chef::DataBagItem.from_hash(encrypted_new_deviceitem)
+      else
+        deviceitem.raw_data = new_deviceitem
+      end
+      deviceitem.create
+      deviceitem.save
+      puts "Saved #{keystore_item_name} to keystore #{keystore_databag_name}"
+
```
Comment (atrull) [inline providers/default.rb:197]: this bit is 403ing, sadly!

**Comment by atrull** [comment]
I'm going to merge this to master and perform the first release, but indeed it needs more testing.

**Pull request merged**